fault-proof/tests/common/process.rs

```diff
@@ -0,0 +1,299 @@
+    game_type: u32,
+    prover_network_rpc: Option<&str>,
+) -> HashMap<String, String> {
+    let mut env = HashMap::new();
+
+    // Required environment variables
+    env.insert("L1_RPC".to_string(), l1_rpc.to_string());
+    env.insert("L1_BEACON_RPC".to_string(), l1_beacon_rpc.to_string());
+    env.insert("L2_RPC".to_string(), l2_rpc.to_string());
+    env.insert("L2_NODE_RPC".to_string(), l2_node_rpc.to_string());
+    env.insert("PRIVATE_KEY".to_string(), private_key.to_string());
+    env.insert("FACTORY_ADDRESS".to_string(), factory_address.to_string());
+    env.insert("GAME_TYPE".to_string(), game_type.to_string());
+
+    // Optional prover network RPC
+    if let Some(prover_rpc) = prover_network_rpc {
+        env.insert("PROVER_NETWORK_RPC".to_string(), prover_rpc.to_string());
+    }
+
+    // Enable info logging
+    env.insert("RUST_LOG".to_string(), "info".to_string());
+
+    // Test-specific configuration for faster game creation
+    env.insert("PROPOSAL_INTERVAL_IN_BLOCKS".to_string(), "10".to_string()); // Much smaller interval for testing
+    env.insert("FETCH_INTERVAL".to_string(), "2".to_string()); // Check more frequently in tests
+
+    env
+}
+
+/// Generate environment variables for the challenger
+#[allow(clippy::too_many_arguments)]
+pub fn generate_challenger_env(
+    l1_rpc: &str,
+    l2_rpc: &str,
+    l2_node_rpc: &str,
+    l1_beacon_rpc: &str,
+    private_key: &str,
+    factory_address: &str,
+    game_type: u32,
+    prover_network_rpc: Option<&str>,
+    malicious_percentage: Option<f64>,
+) -> HashMap<String, String> {
+    let mut env = HashMap::new();
+
+    // Required environment variables
+    env.insert("L1_RPC".to_string(), l1_rpc.to_string());
+    env.insert("L1_BEACON_RPC".to_string(), l1_beacon_rpc.to_string());
+    env.insert("L2_RPC".to_string(), l2_rpc.to_string());
+    env.insert("L2_NODE_RPC".to_string(), l2_node_rpc.to_string());
+    env.insert("PRIVATE_KEY".to_string(), private_key.to_string());
+    env.insert("FACTORY_ADDRESS".to_string(), factory_address.to_string());
+    env.insert("GAME_TYPE".to_string(), game_type.to_string());
+
+    // Optional prover network RPC
+    if let Some(prover_rpc) = prover_network_rpc {
+        env.insert("PROVER_NETWORK_RPC".to_string(), prover_rpc.to_string());
+    }
+
+    // Optional malicious challenge percentage
+    if let Some(percentage) = malicious_percentage {
+        env.insert("MALICIOUS_CHALLENGE_PERCENTAGE".to_string(), percentage.to_string());
+    }
+
+    // Enable info logging
+    env.insert("RUST_LOG".to_string(), "info".to_string());
+
+    // Test-specific configuration
+    env.insert("FETCH_INTERVAL".to_string(), "2".to_string()); // Check more frequently in tests
+    env.insert("MAX_GAMES_TO_CHECK_FOR_CHALLENGE".to_string(), "10".to_string()); // Check more games
+
+    env
+}
+
+/// Wait for a process to produce a specific log line
+#[allow(dead_code)]
+pub async fn wait_for_log_line(
```

Comment:
unused, this also looks like it would be hard to use anyway. Imagine changing the logging a little bit and breaking your test 😬